core/modules/util/StringToVector.cc

```diff
@@ -0,0 +1,84 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2020 LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "util/StringToVector.h"
+
+// System headers
+
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+namespace {
+
+LOG_LOGGER _log = LOG_GET("lsst.qserv.util.StringToVector");
+
+}
+
+namespace lsst {
+namespace qserv {
+namespace util {
+
+std::vector<std::string> splitString(std::string const& original, std::string const& separator) {
```

Comment:
This implementation is too complex. There are much simple and straightforward alternatives like:
```
#include <string>
#include <sstream>
#include <algorithm>
#include <iterator>
 
template <class Container>
Container split2(const std::string& str, char delim=':') {
    std::stringstream ss(str);
    std::string token;
    Container cont;
    while (std::getline(ss, token, delim)) {
        cont.push_back(token);
    }
    return cont;
}
```
Besides, you may also do type conversions into a desired numeric type right within this loop. So that you could return a collection of numeric values right here. And you won't need two separate functions `splitString` and `getIntVectFromStr`.
You may look for other ideas here: http://www.martinbroadhurst.com/how-to-split-a-string-in-c.html

Reply:
I tried this out and it has some bad behaviors. It fails unit tests that the other one passes. The one I'm really concerned about is this one. getline comes back with 0 tokens instead of 1. 
```
    {
        auto vect = util::splitString<std::vector<std::string>>("qsa4$3", ':');
        BOOST_CHECK(vect.size() == 1);
        BOOST_CHECK(vect[0] == "qsa4$3");
    }
```